src/routes/index.js

```diff
@@ -7,6 +8,129 @@ const router = Router();
 // Configurar los routers
 // Ejemplo: router.use('/auth', authRouter);
 
+const consulproduct = async () => {
+  
+  return await Product.findAll({
+      include: {
+          model: Category,
+          attributes: ["name"],
+        through: {
+              attributes: [],
+          }, 
+      }
+  })
+
+} 
+
+const consulUser = async () => {
+  
+  return await User.findAll
+
+} 
+
+router.get('/product', async(req, res) => {
+  
+  const product_name  = req.query.productname
+  let articulo = await consulproduct();
+  
+ 
+  if (product_name) {
+
+      let buscararticulo = await articulo.filter(e => e.name.toLowerCase().includes(product_name.toLowerCase()))
+      buscararticulo.length ? res.status(200).send(buscararticulo) : res.status(404).send("No existe articulo")
+  }
+  
+  else {
+    
+      return res.status(200).send(articulo)
+      
+
+
+  }
+
+   })   
+  
+   router.post('/product', async function (req, res) {
+    try {
+     const { product_name, category, price,description, stock, images, rating, comments, options} = req.body;
+     let newProduct = await Product.create({
+            product_name,
+            price,description,
+            stock,
+            images,
+            rating,
+            comments,
+            options
+                                       
+         })
+
+                        
+
+  let tempbase = await Category.findAll({
+         where: {
+            name : category
```

Comment:
recuerda que category es un array
creo que deberias usar el Op

where: {
 name: { [Op.in]: category}
}

si no sabes que es el Op escribeme al WSP